src/trainer.py

```diff
@@ -1115,14 +1137,67 @@ def _save_checkpoint(self, training_state, phase="pretrain", new_best_macro=Fals
                 "metric_state_{}_epoch_{}{}.th".format(phase, epoch, best_str),
             ),
         )
-        log.info("Saved to %s", self._serialization_dir)
+        log.info("Saved general checkpoints to %s", self._serialization_dir)
```

Comment:
I don't follow the logic here. We always save exactly one checkpoint, either the general ones for pretraining or the task-specific one in target task training.